### PR TITLE
[QD Reconfig] 4. Install reconfig observer in TransactionOrchestrator

### PR DESCRIPTION
**Pull request by longbowlu**
This PR installs OnsiteReconfigObserver in TransactionOrchestrator that reconfig QuorumDriver

### CODE REVIEW

File: crates/sui/tests/transaction_orchestrator_tests.rs [hunk 1]
```diff
@@ -137,6 +154,116 @@ async fn test_fullnode_wal_log() -> Result<(), anyhow::Error> {
...
+                // Retry a transaction if it times out (e.g. in epoch boundary) after 500 ms.
+                tokio::time::sleep(tokio::time::Duration::from_millis(500)).await;
+            }
+            // Wait for a little bit before getting to the next tx.
+            tokio::time::sleep(tokio::time::Duration::from_millis(100)).await;
+        }
+    });
+    
+
+    // uncomment this will cause fullnode not be able to advance epoch
```
Comment (longbowlu) [inline crates/sui/tests/transaction_orchestrator_tests.rs:250]: ^^

File: crates/sui-core/src/checkpoints/checkpoint_executor/mod.rs [hunk 2]
```diff
@@ -40,7 +40,7 @@ use tracing::{debug, error, info, warn};
 use typed_store::{rocks::TypedStoreError, Map};
 
 use crate::{
-    authority::{AuthorityState, EffectsNotifyRead},
+    authority::{authority_store::EffectsStore, AuthorityState, EffectsNotifyRead},
```
Comment (longbowlu) [inline crates/sui-core/src/checkpoints/checkpoint_executor/mod.rs:43]: will move changes in this file into a separate PR

File: crates/sui-node/src/lib.rs [hunk 3]
```diff
@@ -216,12 +206,14 @@ impl SuiNode {
...
-            Some(Arc::new(TransactiondOrchestrator::new(
-                Arc::new(arc_net),
-                state.clone(),
-                config.db_path(),
-                &prometheus_registry,
-            )))
+            Some(Arc::new(
+                TransactiondOrchestrator::new_with_network_clients(
+                    state.clone(),
+                    checkpoint_executor_handle.subscribe_to_end_of_epoch(),
```
Comment (lxfind) [inline crates/sui-node/src/lib.rs:212]: Could you add a TODO here to replace it with a layer channel?

File: crates/sui/tests/transaction_orchestrator_tests.rs (hunk 1 above)
Comment (lxfind) [reply crates/sui/tests/transaction_orchestrator_tests.rs:256]: Is this a bug tracked somewhere?

Comment (longbowlu) [reply crates/sui/tests/transaction_orchestrator_tests.rs:256]: @mystenmark  and @williampsmith have been looking into it

Comment (williampsmith) [reply crates/sui/tests/transaction_orchestrator_tests.rs:256]: Out of curiosity, does this bug remain after rebasing now? There were 2 semi-related bug fixes recently merged

**Review by lxfind — APPROVED** [review]

File: crates/sui-core/src/checkpoints/checkpoint_executor/mod.rs (hunk 2 above)
Comment (longbowlu) [reply crates/sui-core/src/checkpoints/checkpoint_executor/mod.rs]: handled in https://github.com/MystenLabs/sui/pull/7177
